### PR TITLE
ShardedDaemonProcess: Use Interger in javadsl function

### PR DESCRIPTION
**Pull request by octonato**
The new `init` method is using `Int` instead of `Interger` and it doesn't play well when using from Java code.

### CODE REVIEW

**Comment by johanandren** [comment]
D'oh. Has to be changed in both Java DSL API and impl.

**Comment by renatocaval** [comment]
> D'oh. Has to be changed in both Java DSL API and impl.

Fixed now

**Comment by johanandren** [comment]
And a mima exclude for changing the signature (which nobody can have called anyway, right?)

**Comment by renatocaval** [comment]
> And a mima exclude for changing the signature (which nobody can have called anyway, right?)

Yep, forgot this as well. 

It's possible to call it though, see https://github.com/akka/akka-projection/pull/237/files#diff-15f1e23685faaa060a8688a3a85b55caR240

If someone uses it, the impact will be minimal. In Akka 2.6.7, it will infer it to Integer and then the user will be casting to Integer eventually. 

BTW, that API is marked as @ApiMayChange.

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala [hunk 1]
```diff
@@ -168,7 +168,7 @@ private[akka] final class ShardedDaemonProcessImpl(system: ActorSystem[_])
       messageClass: Class[T],
       name: String,
       numberOfInstances: Int,
-      behaviorFactory: function.Function[Int, Behavior[T]],
+      behaviorFactory: function.Function[Integer, Behavior[T]],
```
Comment (patriknw) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala:171]: by the way, we should use `java.util.function.IntFunction` instead

**Review by johanandren — APPROVED** [review]

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala (hunk 1 above)
Comment (renatocaval) [reply akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala:171]: > by the way, we should use java.util.function.IntFunction instead

Ok, I will change it on another PR

**Comment by renatocaval** [comment]
@patriknw, I see that you re-added yourself as reviewer. Is that because you prefer to have `IntFunction` in this PR? 

Your first review is still valid which makes this PR mergeable.

Anyway, I will wait for you to merge or not. In the meantime, I'm changing it to `IntFunction`. May push here or send another PR.

**Comment by renatocaval** [comment]
ok, `IntFunction` is now integrated on this PR.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**